### PR TITLE
More optimized redaction of build events

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch 

**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/731, https://github.com/buildbuddy-io/buildbuddy-internal/issues/661

### CODE REVIEW

File: enterprise/server/auth/auth.go [hunk 1]
```diff
@@ -534,6 +534,10 @@ func (a *OpenIDAuthenticator) ParseAPIKeyFromString(input string) string {
 }
 
 func (a *OpenIDAuthenticator) AuthContextFromAPIKey(ctx context.Context, apiKey string) context.Context {
+	if _, ok := ctx.Value(APIKeyHeader).(string); ok {
+		log.Warningf("Overwriting existing value of %q in context.", APIKeyHeader)
+	}
+	ctx = context.WithValue(ctx, APIKeyHeader, apiKey)
```
Comment (tylerwilliams) [inline enterprise/server/auth/auth.go:540]: Will this trigger in normal operation?

File: server/util/redact/redact.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	"github.com/golang/protobuf/proto"
+
+	bespb "github.com/buildbuddy-io/buildbuddy/proto/build_event_stream"
+)
+
+const (
+	RedactionFlagAPIKey = 1 << iota
+)
+
+func RedactAPIKey(ctx context.Context, env environment.Env, event *bespb.BuildEvent) error {
```
Comment (tylerwilliams) [inline server/util/redact/redact.go:18]: I am more guilty than most of naming stuff like this -- but the "right" way to name this in go is probably to call this method APIKey() and then your code would read:

if err := redact.ApiKey(asdasd); err != nil {
return err
}

File: server/util/redact/redact.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+	contextKey, ok := ctx.Value("x-buildbuddy-api-key").(string)
+	if ok {
+		txt = strings.ReplaceAll(txt, contextKey, "<REDACTED>")
+	}
+
+	return proto.UnmarshalText(txt, event)
+}
+
+func getConfiguredAPIKey(env environment.Env) string {
```
Comment (tylerwilliams) [inline server/util/redact/redact.go:72]: can you add a comment on this / possibly also name it differently so it's clear what this API key is? (it's not the one sent in the request)

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: enterprise/server/auth/auth.go (hunk 1 above)
Comment (bduffany) [reply enterprise/server/auth/auth.go]: Nope -- switched to `alert.UnexpectedEvent`

File: server/util/redact/redact.go (hunk 2 above)
Comment (bduffany) [reply server/util/redact/redact.go]: sg, done. I think it looks kind of weird within the package itself, but it's probably better to make it more readable at the callsite.

File: server/util/redact/redact.go (hunk 3 above)
Comment (bduffany) [reply server/util/redact/redact.go]: Done

File: server/util/redact/redact.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/golang/protobuf/proto"
+
+	bespb "github.com/buildbuddy-io/buildbuddy/proto/build_event_stream"
+)
+
+const (
+	RedactionFlagAPIKey = 1 << iota
```
Comment (tylerwilliams) [inline server/util/redact/redact.go:15]: Since this is written into the database -- can you explicitly define it?

**Review by tylerwilliams — APPROVED** [review]
LGTM

File: server/util/redact/redact.go (hunk 4 above)
Comment (bduffany) [reply server/util/redact/redact.go]: Done

**Pull request merged**